Content.Server/Stack/StackSystem.cs

```diff
@@ -87,24 +88,89 @@ public EntityUid Spawn(int amount, StackPrototype prototype, EntityCoordinates s
         }
 
         /// <summary>
-        ///     Say you want to spawn 97 stacks of something that has a max stack count of 30.
+        ///     Say you want to spawn 97 units of something that has a max stack count of 30.
         ///     This would spawn 3 stacks of 30 and 1 stack of 7.
         /// </summary>
-        public List<EntityUid> SpawnMultiple(string entityPrototype, int amount, EntityCoordinates spawnPosition)
+        public List<EntityUid> SpawnMultiple(int amount, MaterialPrototype materialProto, EntityCoordinates coordinates)
         {
-            var proto = _prototypeManager.Index<EntityPrototype>(entityPrototype);
-            proto.TryGetComponent<StackComponent>(out var stack);
-            var maxCountPerStack = GetMaxCount(stack);
-            var spawnedEnts = new List<EntityUid>();
+            var list = new List<EntityUid>();
+            if (amount <= 0)
+                return list;
+
+            // At least 1 is being spawned, we'll use the first to extract otherwise inaccessible information
+            // ??TODO??: Indexing the entity proto and extracting from its component registry could possibly be better?
+            // it doesn't look like it would save LOC even compressing this to a single loop and I'm not sure what other issues it might introduce
+            var firstSpawn = Spawn(materialProto.StackEntity, coordinates);
+
+            if (!TryComp<StackComponent>(firstSpawn, out var stack) || stack.StackTypeId == null)
+                return list;
```

Comment:
Is there a test that asserts that every material prototype entity has a stack component?

If there isn't this should definitely log errors & properly handle deleting the spawned entity.

Reply:
I wrote a test